### DISCUSSION TITLE
Point to another Acme CA (Buypass)

### DISCUSSION THREAD

**Discussion by henrikrox** [shibayan/appservice-acmebot#300] [Q&A] [answered]
Hi

Amazing bot you have created, a quick question, i want to test with another acme endpoint because of strict use of where the CA recides in the world.

https://community.buypass.com/t/63d4ay/buypass-go-ssl-endpoints-updated-14-05-2020
https://community.buypass.com/t/k9r5cx/get-started
https://api.buypass.com/acme/directory

I tried just changing Acmebot:Endpoint under app settings on the function app, but i guess more needs to be tweaked?

**Comment by henrikrox** [comment, marked as answer]
I see under AcmeBotOptions.cs 


```
namespace AppService.Acmebot.Options
{
  public class AcmebotOptions
  {
    [Required]
    public string Endpoint { get; set; } = "https://acme-v02.api.letsencrypt.org/";
  }
```

Guess i need to change here also?

**Reply by henrikrox** [reply]
nevermind seems it mentioned other places as well, maybe a bigger task to point to another CA. Im using Terraform and WEBSITE_RUN_FROM_PACKAGE

```
app_settings = merge({
    "APPLICATIONINSIGHTS_CONNECTION_STRING" = azurerm_application_insights.insights.connection_string
    "FUNCTIONS_WORKER_RUNTIME"              = "dotnet"
    "WEBSITE_RUN_FROM_PACKAGE"              = "https://shibayan.blob.core.windows.net/azure-appservice-letsencrypt/v3/latest.zip"
  }, local.acmebot_app_settings)
```

```
    public override void Configure(IFunctionsHostBuilder builder)
    {
      FunctionsHostBuilderContext context = builder.GetContext();
      IConfigurationSection section = context.get_Configuration().GetSection("Acmebot");
      ((OptionsBuilder<AcmebotOptions>) OptionsBuilderDataAnnotationsExtensions.ValidateDataAnnotations<AcmebotOptions>(OptionsBuilderConfigurationExtensions.Bind<AcmebotOptions>(OptionsServiceCollectionExtensions.AddOptions<AcmebotOptions>(builder.get_Services()), ConfigurationExtensions.Exists(section) ? (IConfiguration) section : (IConfiguration) context.get_Configuration().GetSection("LetsEncrypt")))).PostConfigure((Action<AcmebotOptions>) (options =>
      {
        if (!(options.Endpoint == "https://acme-v02.api.letsencrypt.org/"))
          return;
        AcmebotOptions acmebotOptions = options;
        if (acmebotOptions.PreferredChain != null)
          return;
        string str;
        acmebotOptions.PreferredChain = str = "DST Root CA X3";
      }));
```

**Reply by shibayan** [reply]
Since the endpoint information is cached, you will need to delete the `D:\home\.acme` directory and start over.

**Reply by henrikrox** [reply]
ok @shibayan but do i only need to change the variable in terraform and start from scratch, or do i need to decompile the dlls and change it from https://shibayan.blob.core.windows.net/azure-appservice-letsencrypt/v3/latest.zip

```
variable "acme_endpoint" {
  type        = string
  description = "Certification authority ACME Endpoint."
  default     = "https://acme-v02.api.letsencrypt.org/"
}
```

**Reply by shibayan** [reply]
Only terraform change.

**Reply by henrikrox** [reply]
Thanks @shibayan worked like a charm, when corona is over, and you are in norway i need to buy you a beer :)
